From 451d7ca09c904aed9c28469331e0cd768f3854a0 Mon Sep 17 00:00:00 2001
From: yuyue <yuyue@mprc.pku.edu.cn>
Date: Wed, 29 Feb 2012 13:20:46 +0800
Subject: [PATCH 312/641] UniCore64: fixed adr error

adr cannot get the address in another section, becuase the offset
is too large. we replace it with ldd instruction.

Signed-off-by: yuyue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/boot/boot.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/boot/boot.S b/arch/unicore64/boot/boot.S
index 07e1cdc..7815582 100644
--- a/arch/unicore64/boot/boot.S
+++ b/arch/unicore64/boot/boot.S
@@ -19,7 +19,7 @@ _start:
 	 * Prepare early devtree for fpga board.
 	 */
 	dmovl	r0, #UC64_PM_DTB_START
-	adr	r1, dtb_data
+	ldd	r1, =dtb_data
 	dmovl	r2, #dtb_data_end - dtb_data
 	call	memcpy
 #endif /* CONFIG_ARCH_FPGA */
-- 
1.7.9.5

